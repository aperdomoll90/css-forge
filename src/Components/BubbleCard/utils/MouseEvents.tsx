export const rotateElement = (event: MouseEvent, element: Element | null) => {
  // get mouse position
  const x = event.clientX
  const y = event.clientY
  // find the middle
  const middleX = window.innerWidth / 2
  const middleY = window.innerHeight / 2

  //get offset from middle
  const offsetX = ((x - middleX) / middleX) * 25
  const offsetY = ((y - middleY) / middleY) * 25
  console.log('element', element)
  if (element) {
    element.style.setProperty('--rotateX', -1 * offsetY + 'deg')
    element.style.setProperty('--rotateY', offsetX + 'deg')
  }
}
