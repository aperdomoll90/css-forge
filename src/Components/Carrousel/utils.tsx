import * as React from 'react'


function getPrevElement(list:any) {
    const sibling = list[0].previousElementSibling
  
    if (sibling instanceof HTMLElement) {
      return sibling
    }
  
    return sibling
  }
  
  function getNextElement(list:any) {
    const sibling = list[list.length - 1].nextElementSibling
  
    if (sibling instanceof HTMLElement) {
      return sibling
    }
  
    return null
  }

  export default function usePosition(ref) {
    const [prevElement, setPrevElement] = React.useState(null)
    const [nextElement, setNextElement] = React.useState(null)
  
    React.useEffect(() => {
      const element = ref.current
  
      const update = () => {
        const rect = element.getBoundingClientRect()
  
        const visibleElements:any  = Array.from(element.children).filter(child => {
          const childRect = child.getBoundingClientRect()
  
          return childRect.left >= rect.left && childRect.right <= rect.right
        })
  
        if (visibleElements.length > 0) {
          setPrevElement(getPrevElement(visibleElements))
          setNextElement(getNextElement(visibleElements))
        }
      }
  
      update()
  
      element.addEventListener('scroll', update, { passive: true })
  
      return () => {
        element.removeEventListener('scroll', update, { passive: true })
      }
    }, [ref])
  
    const scrollToElement = React.useCallback(
      element => {
        const currentNode = ref.current
  
        if (!currentNode || !element) return
  
        let newScrollPosition
  
        newScrollPosition =
          element.offsetLeft +
          element.getBoundingClientRect().width / 2 -
          currentNode.getBoundingClientRect().width / 2
  
        currentNode.scroll({
          left: newScrollPosition,
          behavior: 'smooth',
        })
      },
      [ref]
    )
  
    const scrollRight = React.useCallback(
      () => scrollToElement(nextElement),
      [scrollToElement, nextElement]
    )
  
    const scrollLeft = React.useCallback(
      () => scrollToElement(prevElement),
      [scrollToElement, prevElement]
    )
  
    return {
      hasItemsOnLeft: prevElement !== null,
      hasItemsOnRight: nextElement !== null,
      scrollRight,
      scrollLeft,
    }
  }