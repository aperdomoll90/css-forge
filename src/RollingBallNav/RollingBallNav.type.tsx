import { menuItemsArrayPropsTypes } from '../utils/GlobalTypes.types'

export interface RollingBallNavPropsTypes {
  menuItemsArray: menuItemsArrayPropsTypes[]
  height?: number
  width?: number
  primaryColor?: string
  secondaryColor?: string
  hoverColor?: string
  pressColor?: string
  labelColor?: string
}
