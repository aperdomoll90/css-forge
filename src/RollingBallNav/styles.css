.rollingBallNavWrapper {
  width: var(--width);
  height: var(--height);
  border-radius: 1rem;
  margin: 0;
  padding: 0;
  box-sizing: border-box;
  font-family: 'Poppins', sans-serif;
  display: flex;
  justify-content: center;
  align-items: center;
  background: var(--primaryColor);
}

.rollingBallNav {
  width: 95%;
  height: 84%;
  border-radius: 10px;
  position: relative;
  display: flex;
  justify-content: center;
  align-items: center;
  background: #fff;
}

.rollingBallNav ul {
  display: flex;
  width: 350px;
  height: 100%;
  background: red;
}

.rollingBallNav ul li {
  position: relative;
  list-style: none;
  width: 70px;
  height: 70px;
  z-index: 100;
  margin-top: 2rem;
}

.rollingBallNav ul li a {
  width: 100%;
  height: 100%;

  position: relative;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  text-align: center;
  font-weight: 500;
  transition: 0.5s;
}

.rollingBallNavIcon {
  height: var(--height);
  position: relative;
  display: block;
  size: 1.5em;
  text-align: center;
  transition: 0.5s;
}

li.rollingBallNavActive a {
  transform: translateY(calc((var(--height) / 2.5) * -1));
}

.rollingBallNavIcon svg {
  height: 2rem;
  width: 2rem;
  stroke: var(--primaryColor);
}

.rollingBallNavText {
  position: absolute;
  color: var(--labelColor);
  font-size: 0.75em;
  font-weight: 400;
  letter-spacing: 0.05em;
  opacity: 1;
  transform: translateY(20px);
  transition: 0.5s;
}

li.rollingBallNavActive a .rollingBallNavText {
  opacity: 1;
}

.rollingBallNavIndicator {
  width: 70px;
  height: 70px;
  border-radius: 50%;
  border: 6px solid var(--primaryColor);
  position: absolute;
  top: -50%;
  transition: 0.5s;
  background: var(--secondaryColor);
}

.rollingBallNavIndicator::before {
  content: '';
  width: 15px;
  height: 10px;
  position: absolute;
  top: 50%;
  left: -18.9px;
  border-top-right-radius: 30px;
  box-shadow: 2px -5px 0 0 var(--primaryColor);
  background: transparent;
}

.rollingBallNavIndicator::after {
  content: '';
  width: 15px;
  height: 10px;
  position: absolute;
  top: 50%;
  right: -18.9px;
  border-top-left-radius: 30px;
  box-shadow: -2px -5px 0 0 var(--primaryColor);
  background: transparent;
}

.rollingBallNav ul li:nth-child(1).rollingBallNavActive ~ .rollingBallNavIndicator {
  transform: translateX(calc(70px * 0));
}
.rollingBallNav ul li:nth-child(2).rollingBallNavActive ~ .rollingBallNavIndicator {
  transform: translateX(calc(70px * 1));
}
.rollingBallNav ul li:nth-child(3).rollingBallNavActive ~ .rollingBallNavIndicator {
  transform: translateX(calc(70px * 2));
}
.rollingBallNav ul li:nth-child(4).rollingBallNavActive ~ .rollingBallNavIndicator {
  transform: translateX(calc(70px * 3));
}
.rollingBallNav ul li:nth-child(5).rollingBallNavActive ~ .rollingBallNavIndicator {
  transform: translateX(calc(70px * 4));
}

/* @media (max-width: 850px) {
  .rollingBallNavWrapper  {
   transform: scale(80%);
  }
} */
