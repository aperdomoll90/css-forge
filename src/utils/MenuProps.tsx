import { menuItemsArrayPropsTypes } from './GlobalTypes.types'
import { BarIcon, CameraIcon, CodeIcon, GearIcon, HomeIcon, MapIcon, PawIcon, PlanetIcon, ProfileIcon } from '../IconCollection/IconCollection'
export const menuItemsArray: menuItemsArrayPropsTypes[] = [
  {
    label: 'Home',
    icon: HomeIcon,
    link: '#',
  },
  {
    label: 'Settings',
    icon: GearIcon,
    link: '#',
  },
  {
    label: 'Data',
    icon: BarIcon,
    link: '#',
  },
  {
    label: 'Profile',
    icon: ProfileIcon,
    link: '#',
  },
  {
    label: 'Code',
    icon: CodeIcon,
    link: '#',
  },
]

export const extendedMenuItemsArray: menuItemsArrayPropsTypes[] = [
    ...menuItemsArray,
    {
      label: 'Map',
      icon: MapIcon,
      link: '#',
    },
    {
      label: 'Animations',
      icon: CameraIcon,
      link: '#',
    },
    {
      label: 'Pets',
      icon: PawIcon,
      link: '#',
    },
  ]