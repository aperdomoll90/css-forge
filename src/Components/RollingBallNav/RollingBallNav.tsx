import { useEffect } from 'react';
import './styles.css'

declare global {
    namespace JSX {
      interface IntrinsicElements {
        ['ion-icon']: React.DetailedHTMLProps<React.HTMLAttributes<HTMLElement>, HTMLElement>;
      }
    }
}
declare module 'react' {
    interface HTMLAttributes<T> extends AriaAttributes, DOMAttributes<T> {
      // extends React's HTMLAttributes
      name?: string
    }
  }

function RollingBallNav() {
useEffect(()=>{
    const list = document.querySelectorAll('.rollingBallNavList')
    function activeLink(){
        list.forEach((item)=> item.classList.remove('rollingBallNavActive'))
        this.classList.add('rollingBallNavActive')
    }
    list.forEach((item)=>{
        item.addEventListener('click',activeLink)
    })
},[])

  return (   
 <div className="rollingBallNavWrapper">
   <div className="rollingBallNav">
<ul>
    <li className='rollingBallNavList rollingBallNavActive'>
        <a href="#">
            <span className="rollingBallNavIcon"><ion-icon name="home-outline"/></span>
            <span className="rollingBallNavText">Home</span>
        </a>
    </li>
    <li className='rollingBallNavList'>
        <a href="#">
            <span className="rollingBallNavIcon"><ion-icon name="person-outline"/></span>
            <span className="rollingBallNavText">Profile</span>
        </a>
    </li>
    <li className='rollingBallNavList'>
        <a href="#">
            <span className="rollingBallNavIcon"><ion-icon name="chatbubble-outline"/></span>
            <span className="rollingBallNavText">Messages</span>
        </a>
    </li>
    <li className='rollingBallNavList'>
        <a href="#">
            <span className="rollingBallNavIcon"><ion-icon name="camera-outline"/></span>
            <span className="rollingBallNavText">Photos</span>
        </a>
    </li>
    <li className='rollingBallNavList'>
        <a href="#">
            <span className="rollingBallNavIcon"><ion-icon name="settings-outline"/></span>
            <span className="rollingBallNavText">Settings</span>
        </a>
    </li>
    <div className="rollingBallNavIndicator"> </div>
</ul>
   </div>
</div>
  )
}

export default RollingBallNav
