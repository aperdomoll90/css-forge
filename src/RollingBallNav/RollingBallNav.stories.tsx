import { Meta, StoryFn } from '@storybook/react'
import { RollingBallNav } from './RollingBallNav'
import StorybookContainer from '../StorybookContainer'
import { RollingBallNavPropsTypes } from './RollingBallNav.type'
import { menuItemsArray } from '../utils/MenuProps'

export default {
  title: 'Navigation',
  component: RollingBallNav,
  parameters: {
    jest: ['RollingBallNav.test.tsx'],
  },
} as Meta

const RollingBallNavTemplate: StoryFn<RollingBallNavPropsTypes> = (args) => (
  <StorybookContainer>
    <RollingBallNav {...args} />
  </StorybookContainer>
)

export const RollingBallNavComponent = RollingBallNavTemplate.bind({})
RollingBallNavComponent.args = {
  menuItemsArray: menuItemsArray,
}