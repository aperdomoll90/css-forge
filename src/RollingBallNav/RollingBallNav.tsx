import { useEffect } from 'react'
import './styles.css'
import '../utils/GlobalStyles.css'
import { RollingBallNavPropsTypes } from './RollingBallNav.type'
import { menuItemsArrayPropsTypes } from '../utils/GlobalTypes.types'

export const RollingBallNav: React.FC<RollingBallNavPropsTypes> = ({ height, width, menuItemsArray, primaryColor, secondaryColor, hoverColor, pressColor, labelColor }) => {
  useEffect(() => {
    const list: NodeListOf<Element> = document.querySelectorAll('.rollingBallNavList')
    list.forEach(activeItem => {
      activeItem.addEventListener('click', () => {
        list.forEach(listItem => listItem.classList.remove('rollingBallNavActive'))
        activeItem.classList.add('rollingBallNavActive')
      })
    })
  }, [])

  const stylesProps = {
    '--width': width ? `${width}rem` : '26rem',
    '--height': height ? `${height}rem` : '5.4rem',
    '--primaryColor': primaryColor ? primaryColor : '#222327',
    '--secondaryColor': secondaryColor ? secondaryColor : '#29fd53',
    '--labelColor': labelColor ? labelColor : '#222327',
    '--hoverColor': hoverColor ? hoverColor : '#1c2942',
    '--pressColor': pressColor ? pressColor : '#1c2942',
  }

  return (
    <div className='rollingBallNavWrapper' style={stylesProps as React.CSSProperties}>
      <div className='rollingBallNav'>
        <ul>
          {menuItemsArray &&
            menuItemsArray.map((item: menuItemsArrayPropsTypes) => {
              console.log('item', item)
              return (
                <li className='rollingBallNavList'>
                  {/* <a href={item.link}> */}
                  <a>
                    <div className='rollingBallNavIcon'> <>{item.icon}</></div>
                    <span className='rollingBallNavText'>{item.label}</span>
                  </a>
                </li>
              )
            })}
          <div className='rollingBallNavIndicator'> </div>
        </ul>
      </div>
    </div>
  )
}
