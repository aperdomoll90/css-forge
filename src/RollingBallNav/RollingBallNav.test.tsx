import { render } from '@testing-library/react'
import { RollingBallNav } from './RollingBallNav'
import { menuItemsArray } from '../utils/MenuProps'

describe('RollingBallNav component', () => {
  const props = {
    menuItemsArray: menuItemsArray,
    height: 3,
    width: 5,
    primaryColor: 'rgb(0, 0, 0)',
    secondaryColor: 'rgb(0, 0, 0)',
    hoverColor: 'rgb(0, 0, 0)',
    pressColor: 'rgb(0, 0, 0)',
    labelColor: 'rgb(0, 0, 0)',
  }
  const renderComponent = () => render(<RollingBallNav {...props} />)
  it('should match snapshots', () => {
    const { container } = renderComponent()
    expect(container).toMatchSnapshot()
  })
})
