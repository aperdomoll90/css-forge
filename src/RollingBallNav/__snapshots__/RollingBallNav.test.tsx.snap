// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RollingBallNav component should match snapshots 1`] = `
<div>
  <div
    class="rollingBallNavWrapper"
    style="--width: 5rem; --height: 3rem; --primaryColor: rgb(0, 0, 0); --secondaryColor: rgb(0, 0, 0); --labelColor: rgb(0, 0, 0); --hoverColor: rgb(0, 0, 0); --pressColor: rgb(0, 0, 0);"
  >
    <div
      class="rollingBallNav"
    >
      <ul>
        <li
          class="rollingBallNavList"
        >
          <a>
            <div
              class="rollingBallNavIcon"
            >
               
              <svg
                id="Icon-Home"
                viewBox="0 0 512 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <title>
                  Home
                </title>
                <path
                  class="icon-home"
                  d="M80 212v236a16 16 0 0016 16h96V328a24 24 0 0124-24h80a24 24 0 0124 24v136h96a16 16 0 0016-16V212"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                />
                <path
                  class="icon-home"
                  d="M480 256L266.89 52c-5-5.28-16.69-5.34-21.78 0L32 256M400 179V64h-48v69"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                />
              </svg>
            </div>
            <span
              class="rollingBallNavText"
            >
              Home
            </span>
          </a>
        </li>
        <li
          class="rollingBallNavList"
        >
          <a>
            <div
              class="rollingBallNavIcon"
            >
               
              <svg
                id="Icon-Gear"
                viewBox="0 0 512 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <title>
                  Settings
                </title>
                <path
                  class="icon-gear"
                  d="M262.29 192.31a64 64 0 1057.4 57.4 64.13 64.13 0 00-57.4-57.4zM416.39 256a154.34 154.34 0 01-1.53 20.79l45.21 35.46a10.81 10.81 0 012.45 13.75l-42.77 74a10.81 10.81 0 01-13.14 4.59l-44.9-18.08a16.11 16.11 0 00-15.17 1.75A164.48 164.48 0 01325 400.8a15.94 15.94 0 00-8.82 12.14l-6.73 47.89a11.08 11.08 0 01-10.68 9.17h-85.54a11.11 11.11 0 01-10.69-8.87l-6.72-47.82a16.07 16.07 0 00-9-12.22 155.3 155.3 0 01-21.46-12.57 16 16 0 00-15.11-1.71l-44.89 18.07a10.81 10.81 0 01-13.14-4.58l-42.77-74a10.8 10.8 0 012.45-13.75l38.21-30a16.05 16.05 0 006-14.08c-.36-4.17-.58-8.33-.58-12.5s.21-8.27.58-12.35a16 16 0 00-6.07-13.94l-38.19-30A10.81 10.81 0 0149.48 186l42.77-74a10.81 10.81 0 0113.14-4.59l44.9 18.08a16.11 16.11 0 0015.17-1.75A164.48 164.48 0 01187 111.2a15.94 15.94 0 008.82-12.14l6.73-47.89A11.08 11.08 0 01213.23 42h85.54a11.11 11.11 0 0110.69 8.87l6.72 47.82a16.07 16.07 0 009 12.22 155.3 155.3 0 0121.46 12.57 16 16 0 0015.11 1.71l44.89-18.07a10.81 10.81 0 0113.14 4.58l42.77 74a10.8 10.8 0 01-2.45 13.75l-38.21 30a16.05 16.05 0 00-6.05 14.08c.33 4.14.55 8.3.55 12.47z"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                />
              </svg>
            </div>
            <span
              class="rollingBallNavText"
            >
              Settings
            </span>
          </a>
        </li>
        <li
          class="rollingBallNavList"
        >
          <a>
            <div
              class="rollingBallNavIcon"
            >
               
              <svg
                id="Icon-bar"
                viewBox="0 0 512 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <title>
                  Chart
                </title>
                <path
                  class="icon-bar"
                  d="M32 32v432a16 16 0 0016 16h432"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                />
                <rect
                  class="icon-bar-bars"
                  height="192"
                  rx="20"
                  ry="20"
                  stroke="black"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  width="80"
                  x="96"
                  y="224"
                />
                <rect
                  class="icon-bar-bars"
                  height="240"
                  rx="20"
                  ry="20"
                  stroke="black"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  width="80"
                  x="240"
                  y="176"
                />
                <rect
                  class="icon-bar-bars"
                  height="304"
                  rx="20"
                  ry="20"
                  stroke="black"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  width="80"
                  x="383.64"
                  y="112"
                />
              </svg>
            </div>
            <span
              class="rollingBallNavText"
            >
              Data
            </span>
          </a>
        </li>
        <li
          class="rollingBallNavList"
        >
          <a>
            <div
              class="rollingBallNavIcon"
            >
               
              <svg
                id="Icon-profile"
                viewBox="0 0 512 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <title>
                  Person
                </title>
                <path
                  class="icon-profile"
                  d="M344 144c-3.92 52.87-44 96-88 96s-84.15-43.12-88-96c-4-55 35-96 88-96s92 42 88 96z"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                />
                <path
                  class="icon-profile"
                  d="M256 304c-87 0-175.3 48-191.64 138.6C62.39 453.52 68.57 464 80 464h352c11.44 0 17.62-10.48 15.65-21.4C431.3 352 343 304 256 304z"
                  stroke-miterlimit="10"
                />
              </svg>
            </div>
            <span
              class="rollingBallNavText"
            >
              Profile
            </span>
          </a>
        </li>
        <li
          class="rollingBallNavList"
        >
          <a>
            <div
              class="rollingBallNavIcon"
            >
               
              <svg
                id="Icon-code"
                viewBox="0 0 512 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <title>
                  Code Working
                </title>
                <circle
                  cx="256"
                  cy="256"
                  r="26"
                />
                <circle
                  cx="346"
                  cy="256"
                  r="26"
                />
                <circle
                  cx="166"
                  cy="256"
                  r="26"
                />
                <path
                  class="icon-code"
                  d="M160 368L32 256l128-112M352 368l128-112-128-112"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                />
              </svg>
            </div>
            <span
              class="rollingBallNavText"
            >
              Code
            </span>
          </a>
        </li>
        <div
          class="rollingBallNavIndicator"
        >
           
        </div>
      </ul>
    </div>
  </div>
</div>
`;
